prototyping/xarray/xdose.py

```diff
@@ -0,0 +1,198 @@
+from typing import Tuple
+
+import numpy as np
+import pydicom
+import xarray as xr
+
+from pymedphys._dicom import constants, coords, dose
+
+XARRAY_DIMS_BY_COORD_SYSTEM_AND_ORIENT = {
+    "S": ["y", "z", "x"],
+    "P": ["y", "z", "x"],
+    "P dec": ["y", "x", "z"],
+    "D": ["z", "y", "x"],
+    "D dec": ["z", "x", "y"],
+}
+
+
+def xdose_from_dataset(ds, coord_system="SUPPORT", decimals_to_round_coords=-1):
+
+    dose_values = dose.dose_from_dataset(ds)
+    xr_dims = xarray_dims_from_dataset(ds, coord_system=coord_system)
+    xr_coords = xarray_coords_from_dataset(
+        ds,
+        coord_system=coord_system,
+        round_to_decimals=decimals_to_round_coords,
+    )
+
+    return xr.DataArray(
+        data=dose_values,
+        dims=xr_dims,
+        coords=xr_coords,
+        name="dose",
+        attrs={"units": ds.DoseUnits.title(), "coord_system": coord_system},
+    )
+
+
+def coords_from_dataset(
+    ds: "pydicom.dataset.Dataset", coord_system: str = "SUPPORT"
```

Comment:
Can use the "Literal" type here for coord_system

Reply:
Yup, `typing_extension` is the way to go until we drop support for Python 3.6/3.7